OpenFeature/src/main/java/dev/openfeature/sdk/async/Extensions.kt

```diff
@@ -13,21 +14,36 @@ import kotlinx.coroutines.flow.take
 import kotlinx.coroutines.launch
 import kotlinx.coroutines.suspendCancellableCoroutine
 
-fun OpenFeatureClient.toAsync(dispatcher: CoroutineDispatcher = Dispatchers.IO): AsyncClient {
-    return AsyncClientImpl(this, dispatcher)
+fun OpenFeatureClient.toAsync(
+    dispatcher: CoroutineDispatcher = Dispatchers.IO
+): AsyncClient {
+    val provider = OpenFeatureAPI.getProvider()
+    requireNotNull(provider)
+    return AsyncClientImpl(
+        this,
+        provider,
+        dispatcher
+    )
 }
 
-internal fun observeProviderReady(
+internal fun ProviderEventObserver.observeProviderReady(
     dispatcher: CoroutineDispatcher = Dispatchers.IO
-) = EventHandler.eventsObserver(dispatcher)
-    .observe<OpenFeatureEvents.ProviderReady>()
+) = observe<OpenFeatureEvents.ProviderReady>()
     .onStart {
-        if (EventHandler.providerStatus().isProviderReady()) {
+        if (isProviderReady()) {
             this.emit(OpenFeatureEvents.ProviderReady)
         }
     }
 
-suspend fun awaitProviderReady(
+suspend fun OpenFeatureAPI.awaitProviderReady(
```

Comment:
I am thinking if `awaitProviderReady` could be moved away from OpenFeatureAPI (the global layer), and rather something like: `OpenFeatureAPI.getClient().awaitReady`

Reply:
this works too. I give it a try.